clients/java/src/main/java/io/camunda/zeebe/client/impl/http/TypedApiEntityConsumer.java

```diff
@@ -91,13 +93,23 @@ public JsonApiEntityConsumer(
 
     @Override
     public ApiEntity<T> generateContent() throws IOException {
-      buffer.asParserOnFirstToken();
-
-      if (isResponse) {
-        return ApiEntity.of(json.readValue(buffer.asParserOnFirstToken(), type));
+      try {
+        if (isResponse) {
+          return ApiEntity.of(json.readValue(buffer.asParserOnFirstToken(), type));
+        }
+        return ApiEntity.of(json.readValue(buffer.asParserOnFirstToken(), ProblemDetail.class));
+      } catch (final IOException ioe) {
+        // write the original JSON response into an error response
+        final ByteArrayOutputStream output = new ByteArrayOutputStream();
+        final JsonGenerator generator = json.createGenerator(output);
```

Comment:
`JsonGenerator` is closable so may need to be closed
```suggestion
    try (final ByteArrayOutputStream output = new ByteArrayOutputStream();
         final JsonGenerator generator = json.createGenerator(output)) {
```

Reply:
Yep, let's do that